src/map/Map.js

```diff
@@ -1697,6 +1700,12 @@ export var Map = Evented.extend({
 			noUpdate: noUpdate
 		});
 
+		if (!this._tempFireZoomEvent) {
+			this._tempFireZoomEvent = this._zoom !== this._animateToZoom;
+		}
+
+		this._move(this._animateToCenter, this._animateToZoom, undefined, true);

```

Comment:
This is the necessary call to set the `_pixelOrigin` but we need to prevent firing the `zoom` event (to keep the current behaviour)